packages/controller/src/WsServer.ts

```diff
@@ -59,6 +59,20 @@ export default class WsServer {
 		);
 	}
 
+	remoteAddr(req: IncomingMessage) {
+		const remote = req.socket.remoteAddress ?? "";
+		if (
+			remote && req.headers["x-forwarded-for"]
+			&& (
+				(req.socket.remoteFamily === "IPv4" && this.controller.trustedProxies.check(remote, "ipv4"))
+				|| (req.socket.remoteFamily === "IPv6" && this.controller.trustedProxies.check(remote, "ipv6"))
+			)
+		) {
+			return (req.headers["x-forwarded-for"] as string).split(",").slice(-1)[0].trim();
```

Comment:
Thoughts on using a while loop to get the rightmost address that is not trusted?

Reply:
For what gain? Performance? Clarity?

Reply:
The gain would be getting the first address that you don't consider a trusted proxy, which is hopefully the client. This would allow for multiple proxies to be used. Taken from [Selecting an IP address](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-For#selecting_an_ip_address).

Reply:
The code already fetches the rightmost address.
```js
> "forged, 10.0.0.1".split(",").slice(-1)[0].trim()
'10.0.0.1'
```
Here `slice(-1)` returns a slice of the array starting from the last element (negative numbers index from the end.

Reply:
To put it differently, I don't think anyone are operating their controller behind a chain of proxies so I will put off implementing the logic to support that until someone actually show up needing it. I believe this practise is called lazy coding.